src/java/org/apache/cassandra/index/sai/analyzer/AnalyzerEqOperatorSupport.java

```diff
@@ -56,15 +48,26 @@ public class AnalyzerEqOperatorSupport
                                                            "Possible values are %s but found ",
                                                            OPTION, Arrays.toString(Value.values()));
 
-    public static final String EQ_RESTRICTION_ON_ANALYZED_WARNING =
-    String.format("Columns [%%s] are restricted by '=' and have analyzed indexes [%%s] able to process those restrictions. " +
-                  "Analyzed indexes might process '=' restrictions in a way that is inconsistent with non-indexed queries. " +
-                  "While '=' is still supported on analyzed indexes for backwards compatibility, " +
-                  "it is recommended to use the ':' operator instead to prevent the ambiguity. " +
-                  "Future versions will remove support for '=' on analyzed indexes. " +
-                  "If you want to forbid the use of '=' on analyzed indexes now, " +
+    public static final String DEPRECATION_WARNING =
+    String.format("Future versions will remove support for '=' on analyzed indexes preventing this ambiguity. " +
+                  "If you want to forbid the use of '=' on analyzed indexes now and prevent this warning, " +
                   "please use '%s':'%s' in the index options.",
-                  OPTION, Value.UNSUPPORTED.toString().toLowerCase());
+                  AnalyzerEqOperatorSupport.OPTION,
+                  AnalyzerEqOperatorSupport.Value.UNSUPPORTED.toString().toLowerCase());
+
+    public static final String EQ_RESTRICTION_ON_ANALYZED_WARNING =
+    "Columns [%s] are restricted by '=' and have analyzed indexes [%s] able to process those restrictions. " +
+    "Analyzed indexes might process '=' restrictions in a way that is inconsistent with non-indexed queries. " +
+    "While '=' is still supported on analyzed indexes for backwards compatibility, " +
+    "it is recommended to use the ':' operator instead to prevent the ambiguity. " +
+    DEPRECATION_WARNING;
+
+    public static final String LWT_CONDITION_ON_ANALYZED_WARNING =
+    "Index analyzers not applied to LWT conditions on columns [%s]. " +
+    "The LWT conditions use the '=' operator on columns supported by analyzed indexes " +
+    "that process the '=' operator same as the ':' operator in SELECT queries for backwards compatibility. " +
+    "This means that '=' has different meaning in LWT writes (not analyzed) than in SELECT queries (analyzed). " +
+    DEPRECATION_WARNING;
```

Comment:
I was thinking about this more.  Not sure we actually need a warning here.  As a user I don't think I would expect the Analyzer to apply here.  But if we want to keep a warning, not sure it needs to be so long.  Probably just need the first part here:

```suggestion
    public static final String LWT_CONDITION_ON_ANALYZED_WARNING =
    "Index analyzers not applied to LWT conditions on columns [%s]."
```

Reply:
I have shortened the warning message to only include the first phrase. Indeed it had too many explanations :)

As for whether we need a warning at all, I'd prefer to leave it at least in this abbreviated form. Some of us have had doubts about what should be the correct behaviour, so some users might have the same doubts. Even if we conclude the current behaviour is right, the warning can help dissipate any doubts.